app/services/sources/RedditSource.java

```diff
@@ -119,17 +107,34 @@ public RedditClient getRedditClient() {
         return redditClient;
     }
 
-    public List<Submission> getHotPosts() {
+    public List<Submission> getHotPosts(int numPosts) {
         SubredditPaginator sp = new SubredditPaginator(redditClient);
+        Iterator<Listing<Submission>> listingIterator = sp.iterator();
 
-        sp.setLimit(MAX_REQUEST_SIZE);
+        sp.setLimit(numPosts);
         sp.setSorting(Sorting.HOT);
         sp.setTimePeriod(TimePeriod.DAY);
         sp.next(true);
 
-        Listing<Submission> list = sp.getCurrentListing();
+        ArrayList<Submission> hotPosts = new ArrayList<>();
```

Comment:
This is perfectly fine as is, but using `List<Submission>...` allows us to change things much easier later on if we decide to use a different implementation of List.